waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -235,6 +235,31 @@ proc loadFromStorage(pm: PeerManager) =
   else:
     debug "successfully queried peer storage"
 
+proc canBeConnected*(pm: PeerManager,
+                     peerId: PeerId): bool =
+  # Returns if we can try to connect to this peer, based on past failed attempts
+  # It uses an exponential backoff. Each connection attempt makes us
+  # wait more before trying again.
+  let failedAttempts = pm.peerStore[NumberFailedConnBook][peerId]
+
+  # if it never errored, we can try to connect
+  if failedAttempts == 0:
+    return true
+
+  # if there are too many failed attemps, do not reconnect 
+  # the check for 14 comes from overflow described in https://github.com/waku-org/nwaku/issues/1665)
+  if failedAttempts >= pm.maxFailedAttempts or failedAttempts > 14:
```

Comment:
perhaps i would enforce the chech that `maxFailedAttempts` < some_value when we init the peer manager.

note also that with the current values 14 is ok, but if i set `backoffFactor` to 10000 (nonsense perhaps) i will hit the overflow way earlier.